lib/iris/tests/unit/util/test_mask_cube.py

```diff
@@ -40,9 +63,132 @@ def test_mask_cube_2d(self):
-        self.assertTrue(np.all(expected.data.mask == returned.data.mask))
+        cube.data = cube.data.data
+        returned = mask_cube(cube, discontiguity_array, in_place=True)
+        np.testing.assert_array_equal(expected.data.mask, cube.data.mask)
+        self.assertOriginalMetadata(cube, "full2d_global")
+        self.assertIs(returned, None)
+
+    def test_mask_cube_2d_not_in_place(self):
+        # This tests the masking of a 2d data array
+        cube = self.cube_2d
+        discontiguity_array = ma.getmaskarray(cube.data).copy()
+        expected = cube.copy()
+
+        # Remove mask so that we can pass an unmasked data set to
+        # mask_discontiguities, and check that it masks the correct point by
+        # comparing with masked data
+        cube.data = cube.data.data
+        returned = mask_cube(cube, discontiguity_array, in_place=False)
+        np.testing.assert_array_equal(expected.data.mask, returned.data.mask)
+        self.assertOriginalMetadata(returned, "full2d_global")
+        self.assertFalse(ma.is_masked(cube.data))
+
+    def test_mask_cube_lazy_in_place(self):
+        cube = simple_2d()
+        cube.data = cube.lazy_data()
+        mask = [0, 1, 1, 0]
+        returned = mask_cube(cube, mask, in_place=True)
+        self.assertTrue(cube.has_lazy_data())
+        # Touch the data so lazyness status doesn't affect CML check.
+        cube.data
+        self.assertOriginalMetadata(cube, "simple_2d")
+        for subcube in cube.slices("foo"):
+            # Mask should have been broadcast across "bar" dimension.
+            np.testing.assert_array_equal(subcube.data.mask, mask)
+        self.assertIs(returned, None)
+
+
+class TestCoordMask(tests.IrisTest, MaskCubeMixin):
+    """Tests with mask specified as a Coord."""
+
+    def setUp(self):
+        self.cube = simple_2d()
+
+    def test_mask_cube_2d_first_dim(self):
+        mask_coord = iris.coords.AuxCoord([0, 1, 0], long_name="mask", units=1)
+        self.cube.add_aux_coord(mask_coord, 0)
+
+        returned = mask_cube(self.cube, mask_coord, in_place=False)
+        # Remove extra coord so we can check against original metadata.
+        returned.remove_coord(mask_coord)
+        self.assertOriginalMetadata(returned, "simple_2d")
+        for subcube in returned.slices("bar"):
+            # Mask should have been broadcast across "foo" dimension.
+            np.testing.assert_array_equal(subcube.data.mask, mask_coord.points)
+
+    def test_mask_cube_2d_second_dim(self):
+        mask_coord = iris.coords.AuxCoord(
+            [0, 0, 1, 1], long_name="mask", units=1
+        )
+        returned = mask_cube(self.cube, mask_coord, in_place=False, dim=1)
+        self.assertOriginalMetadata(returned, "simple_2d")
+        for subcube in returned.slices("foo"):
+            # Mask should have been broadcast across "bar" dimension.
+            np.testing.assert_array_equal(subcube.data.mask, mask_coord.points)
```

Comment:
@rcomer Did you want to add a simple test here for the case where the `points_to_mask` is a simple `numpy` array?

Bonus credits for passing the `ndarray` but with no `dim` specified and catching the traceback (which hopefully should "just happen" :wink:)

That would then cover the three use cases of the mask either being a `Cube`, `Coord` or a vanilla `ndarray` :+1:

Reply:
The `dim` keyword is only used when passing a coordinate (if the coordinate is not attached to the cube).  For arrays, we just rely on numpy broadcasting.  This case covers broadcasting of an array-like:

https://github.com/rcomer/iris/blob/4fb113a3d11dfc66bd5d7bd6dd17c1d5fc99b53d/lib/iris/tests/unit/util/test_mask_cube.py#L87-L99

I've updated the name of it to make it clearer.

Or is there something else I've missed?

Reply:
Nope, actually it's me misremembering. 

Just clarified with @rcomer that the `dim` kwarg is only required for coordinates, and a vanilla `ndarray` simply relies on `numpy` broadcasting :+1: